src/Servers/Kestrel/Core/src/Internal/Http/Http1OutputProducer.cs

```diff
@@ -273,6 +273,19 @@ public void Advance(int bytes)
         }
     }
 
+    public long UnflushedBytes
+    {
+        get
+        {
+            var bytes = _position + _advancedBytesForChunk + _pipeWriter.UnflushedBytes;
+            for (var i = 0; i < _completedSegments?.Count; i++)
+            {
+                bytes += _completedSegments[i].Length;
+            }
```

Comment:
Is this cheaper than keeping the count?

Reply:
I figured since this is only used when you haven't flushed headers yet it wasn't worth it to add another byte counting field.

Reply:
But doesn't that account for the first write(s) to every HTTP/1.1 response body? That seems very common particularly for small JSON responses. Computers are fast, and given `_completedSegments?.Count` should be 4 or less based on the default flush threshold, you could probably argue this is effectively O(1). But given how frequently `JsonConverter.ShouldFlush()` is called, we should probably keep track as we go.

It's too bad everyone who's not checking `UnflushedBytes` has to pay the (very small) cost of this, but the alternative makes writing in a loop and repeatedly checking `UnflushedBytes` O(n^2) where n is the flush threshold. I think this is more reason to have S.T.J do all the counting itself rather than rely on `PipeWriter.UnflushedBytes`. But in the short term, I think incrementing `UnflushedBytes` as we go is the right move.